### PR TITLE
Update learning_by_example.rst

### PR DESCRIPTION
**Pull request by Kolloom**

### CODE REVIEW

**Comment by Kolloom** [comment]
Never done this three way merge before. Could deleted something during the process. Let me know

**Comment by Kolloom** [comment]
https://github.com/eulertour/manim/pull/24 Not sure if this is still relevent

File: docs/source/installation/mac.rst [hunk 1]
```diff
@@ -1,4 +1,9 @@
 Mac
 ===
 
-A stub for mac installation
+The simplest way to install the system dependencies on Mac OS X is with Homebrew.
+
+1. Installing Cairo: ``brew install cairo``
+2. Installing Sox: ``brew install sox``
+3. Installing ffmpeg: ``brew install ffmpeg``
+4. Installing latex (MiKTeX): https://miktex.org/howto/install-miktex-mac
```
Comment (eulertour) [inline docs/source/installation/mac.rst]: Did you mean to add `pip install manimlib` (or maybe `pip install --user manimlib`) to the end?

Comment (Kolloom) [reply docs/source/installation/mac.rst]: Does mac python install includes pip in the PATH? I had to add ``script/`` to PATH under windows.

**Comment by Kolloom** [comment]
I just realized this patch is more work than it is supposed to be. 

You are not going to like this but I rewrote the learning_by_example.py. 

see the commit msg for changes, your changes should all be addressed. 

I hope the end result is more clear than what we had.

File: docs/source/getting_started/learning_by_example.rst [hunk 2]
```diff
@@ -22,129 +26,98 @@ manim. For instance, take ``SquareToCircle``. Copy the code given below to a new
...
-:class:`~animation.animation.Animation` takes one or more
-:class:`~mobject.mobject.Mobject` instances as arguments, which it animates
-when passed to :meth:`~scene.scene.Scene.play`. This is how video is typically
-created in manim. :class:`~mobject.mobject.Mobject` instances are automatically
-added to the :class:`~scene.scene.Scene` when they are animated. You can add a
-:class:`~mobject.mobject.Mobject` to the :class:`~scene.scene.Scene` manually
-by passing it as an argument to :meth:`~scene.scene.Scene.add`.
+``flip()`` ``rotate()`` ``set_fill()`` apply various modifications to the mobjects before animating
+them.  The call to :meth:`~mobject.mobject.Mobject.flip` flips the
+:class:`~mobject.geometry.Square` across the RIGHT vector.  This is equivalent
```
Comment (Kolloom) [inline docs/source/getting_started/learning_by_example.rst]: I don't know the point of this. Square will show no changes when flip across right.

File: docs/source/getting_started/learning_by_example.rst [hunk 3]
```diff
@@ -2,12 +2,14 @@ Learning by Example
 ===================
 
 You create videos in manim by writing :class:`~scene.scene.Scene` instances.
-``example_scenes.py`` contains a few simple ones that we can use to learn about
-manim. For instance, take ``SquareToCircle``.
+`example_scenes.py <https://github.com/3b1b/manim/blob/master/example_scenes.py>`_. contains a few simple ones that we can use to learn about
+manim. For instance, take ``SquareToCircle``. Copy the code given below to a new file (say test_manim.py) and run it with ``$ manim test_manim.py SquareToCircle -p``.
```
[the remark this answers is not in the document]
Comment (Kolloom) [reply docs/source/getting_started/learning_by_example.rst]: Changed to point out how to run example_scenes.py

File: docs/source/getting_started/learning_by_example.rst [hunk 4]
```diff
@@ -2,12 +2,14 @@ Learning by Example
...
 
 .. code-block:: python
    :linenos:
 
+   from manimlib.imports import *
```
[the remark this answers is not in the document]
Comment (Kolloom) [reply docs/source/getting_started/learning_by_example.rst]: Addressed

File: docs/source/getting_started/learning_by_example.rst [hunk 5]
```diff
@@ -20,50 +22,64 @@ manim. For instance, take ``SquareToCircle``.
...
+      --livestream          Run in streaming mode
+      --to-twitch           Stream to twitch
+      --with-key TWITCH_KEY
+                            Stream key for twitch
 
-.. raw:: html
 
+
+
+.. raw:: html
```
[the remark this answers is not in the document]
Comment (Kolloom) [reply docs/source/getting_started/learning_by_example.rst]: Addressed

File: docs/source/installation/mac.rst (hunk 1 above)
Comment (eulertour) [reply docs/source/installation/mac.rst]: I honestly don't know, as I don't use a mac.

File: docs/source/getting_started/learning_by_example.rst [hunk 6]
```diff
@@ -1,13 +1,19 @@
 Learning by Example
 ===================
 
-You create videos in manim by writing :class:`~scene.scene.Scene` instances.
-``example_scenes.py`` contains a few simple ones that we can use to learn about
-manim. For instance, take ``SquareToCircle``.
+SquareToCircle
+--------------
+
+``example_scenes.py`` contains a few simple ones that we can use to learn about manim.
```
[the remark this answers is not in the document]
Comment (Kolloom) [reply docs/source/getting_started/learning_by_example.rst]: done

**Pull request merged**